src/borg/testsuite/__init__.py

```diff
@@ -235,26 +235,63 @@ def _assert_dirs_equal_cmp(self, diff, ignore_bsdflags=False, ignore_xattrs=Fals
             self._assert_dirs_equal_cmp(sub_diff, ignore_bsdflags=ignore_bsdflags, ignore_xattrs=ignore_xattrs, ignore_ns=ignore_ns)
 
     @contextmanager
-    def fuse_mount(self, location, mountpoint, *options):
-        os.mkdir(mountpoint)
-        args = ['mount', location, mountpoint] + list(options)
-        self.cmd(*args, fork=True)
+    def fuse_mount(self, location, mountpoint=None, *options, **kwargs):
+        if mountpoint is None:
+            mountpoint = tempfile.mkdtemp()
+        else:
+            os.mkdir(mountpoint)
+        if 'fork' not in kwargs:
+            # For a successful mount, `fork = True` is required. The
+            # option to run the command with `fork = False` is still
+            # necessary to test mount failures like attempting to mount
+            # a read-only repo.
+            kwargs['fork'] = True
+        self.cmd('mount', location, mountpoint, *options, **kwargs)
         self.wait_for_mount(mountpoint)
         yield
         umount(mountpoint)
+        self.wait_for_mount(mountpoint, umount=True)
```

Comment:
hmm, is this a bugfix?

isn't `umount()` expected to return **after** the fs is actually umounted?

Reply:
Since [`helpers.fs.umount`](https://github.com/borgbackup/borg/blob/master/src/borg/helpers/fs.py#L286) simply calls `fusermount` or `umount` and doesn't check the result, I thought it wouldn't hurt to add that. I'm not familiar with the internals of either of those commands, so I can't say if the unmount has always succeeded (or failed) when they return. So a "timeout" might not be necessary, but the check for mount state of the mountpoint seemed useful either way.

Reply:
ok, so guess we at least would notice clearly if the umount did not really work. good.